homeassistant/components/shelly/__init__.py

```diff
@@ -142,6 +146,60 @@ async def async_device_setup(
         )
 
 
+async def async_services_setup(
+    hass: HomeAssistant, dev_reg: device_registry.DeviceRegistry
+):
+    """Set up services."""
+
+    async def async_service_ota_update(call: ServiceCall):
+        """Trigger OTA update."""
+        if not (call.data.get(ATTR_DEVICE_ID) or call.data.get(ATTR_AREA_ID)):
+            _LOGGER.warning("OTA update service: no device or area selected")
+            return
+
+        devices = []
+        if call.data.get(ATTR_AREA_ID):
+            for area_id in call.data.get(ATTR_AREA_ID):
+                devices += [
+                    area_dev
+                    for area_dev in device_registry.async_entries_for_area(
+                        dev_reg, area_id
+                    )
+                    if DOMAIN in next(iter(area_dev.identifiers))
+                ]
+
+        if call.data.get(ATTR_DEVICE_ID):
+            for device_id in call.data.get(ATTR_DEVICE_ID):
+                device = dev_reg.async_get(device_id)
+                if not any(device.id == x.id for x in devices):
+                    devices += [device]
+
+        for device in devices:
+            entry_id = next(iter(device.config_entries))
+            entry_data = hass.data[DOMAIN][DATA_CONFIG_ENTRY][entry_id]
+            rest: ShellyDeviceRestWrapper = entry_data[REST]
+            update_data = rest.device.status["update"]
```

Comment:
If REST sensors are not enabled, the status will be the status of the first init of the device which may not reflect it's current state, even a REST sensor is enabled, the update period for it is 60 seconds, so the status may not reflect the correct device state, since it is used to decide if device needs an update, it would be better to first pull the device latest status:
Example can be found here (need to wrap with timeout and catch the error):
https://github.com/home-assistant/core/blob/86176f1bf9ed90650693d608de4255bcbdb2f4a7/homeassistant/components/shelly/__init__.py#L299
```python
await self.device.update_status()
```

Reply:
`async_trigger_ota_update()` use now `async_refresh()` from `DataUpdateCoordinator` to update status data